### DISCUSSION TITLE
Looking to Hire a Developer with Colour Science

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1048] [Q&A] [answered]
Hey there!  I hope this is appropriate to post.  I am seeking a developer who is very familiar with the field - and library - of colour science.  : )

Our needs are ongoing, but start with tasks that are fairly straightforward.  For example, our first task to plot the XYZ media white point in an ICC profile onto a 1931 chromaticity plot.  We have no fixed deadlines as we are developing a solution in private at the moment, so we are very flexible and just want to take the time to get things right and beautiful.

Anyone interested?  Please send me a message.  Looking forward!

- J

**Comment by thomasmansencal** [comment, marked as answer]
Hi @blue-j,

It should not be too hard to read the ICC Profile with Pillow: https://pillow.readthedocs.io/en/stable/reference/ImageCms.html#PIL.ImageCms.CmsProfile and then use the Colour definitions to do the remaining.

**Reply by blue-j** [reply]
@KelSolaar I appreciate your encouragement!  Unfortunately, I may be unable to go the distance.  I was able to get the horseshoe out without labels and axes, and i started to bring in the CIEXYZ values from the media white point of an ICC profile and convert to xy...  We already have media white point data, so we don't need Pillow to extract it (the values shown are from the Adobe RGB (1998) profile).  Is this right so far?  Ok, then I need to set xy on the diagram.... no clue whatsoever how to do that.  The goal is to output an SVG against transparency.  I'm happy to pay anyone a very reasonable rate to execute this, if it's something easy for you!  : )

J

```
import colour
import matplotlib.pyplot as plt
import numpy as np

XYZ = np.array([0.95045, 1, 1.08905])
XYZ_to_xy(XYZ)

colour.plotting.plot_chromaticity_diagram_CIE1931(
	show_spectral_locus=False,
    show_diagram_colours=True,
    title=(
        ""
    ),
    x_ticker=False,
    y_ticker=False,
    x_label="",
    y_label="",
    axes_visible=False,
    background_transparency=True,
    standalone=True,
)

```

**Reply by thomasmansencal** [reply]
No need to pay anyone for that! :)

There are some details on how to do that in the [Tutorial](https://colour.readthedocs.io/en/develop/tutorial.html#convert-to-chromaticity-coordinates).

```python
xy = colour.XYZ_to_xy(np.array([0.95045, 1, 1.08905]))

figure, axes = colour.plotting.plot_chromaticity_diagram_CIE1931(
    show_spectral_locus=False,
    show_diagram_colours=True,
    diagram_opacity=0.25,
    title=None,
    standalone=False,
)

axes.plot(xy[0], xy[1], "o-", markersize=5, color="red")

colour.plotting.render(
    x_ticker=False,
    y_ticker=False,
    x_label=None,
    y_label=None,
    axes_visible=False,
    background_transparency=True,
    filename="custom_diagram.svg"
)
```
![custom_diagram](https://user-images.githubusercontent.com/99779/194259101-6dece584-737a-44da-a560-33f60e35cc52.svg)

**Comment by blue-j** [comment]
You are so generous with your knowledge.  Thank you kindly!  I hasten to share that I am a proud sponsor of Colour Science!  - J

**Reply by thomasmansencal** [reply]
Thanks for the kind words!

**Comment by blue-j** [comment]
@KelSolaar , i see some tag definitions in the [ICC 4.4](https://www.color.org/specification/ICC.1-2022-05.pdf) spec that are interesting:

"Un-normalized **CIEXYZ** values for illuminant (in which Y is in cd/m2)"

"**nCIEXYZ** values of the illuminant of the PCS, encoded as an XYZNumber"

"**nCIEXYZ** of media white point."

They define **nCIEXYZ** as "CIEXYZ values that have been uniformly scaled so that Y = 1,0 for the adopted white"

Does colour-science have a way to support nCIEXYZ as defined above and "CIEXYZ (in which Y is in cd/m2)"?

Thank you in advance for your support!  : )

J

**Reply by blue-j** [reply]
After testing it appears to me that "nCIEXYZ" is the default XYZ for c-s?

- J

**Reply by thomasmansencal** [reply]
Yes, when you pass `XYZ = np.array([0.95045, 1, 1.08905])`, this is essentially what happened, the illuminant as been normalised by dividing by its `Y`.

**Reply by thomasmansencal** [reply]
```python
>>> import colour
>>> colour.sd_to_XYZ(colour.SDS_ILLUMINANTS["D65"], n=683)
array([  9399.33673411,   9889.19756747,  10769.04185839])
>>> XYZ = colour.sd_to_XYZ(colour.SDS_ILLUMINANTS["D65"], n=683)
>>> XYZ / XYZ[1]
array([ 0.95046506,  1.        ,  1.08897024])
```

**Reply by blue-j** [reply]
@KelSolaar i am learning everyday.  So, are you suggesting that these are equivalent?

1.  Un-normalized CIEXYZ values for illuminant (in which Y is in cd/m2)
2. CIEXYZ values that have been uniformly scaled so that Y = 1,0 for the adopted white

It sounds to me like 2 is what c-s is doing, but 1 is not.

**Reply by thomasmansencal** [reply]
One represents absolute values the later relative ones, they serve different purposes. For a lot of applications, relative values are enough.

**Reply by blue-j** [reply]
@KelSolaar Apologies for pushing my luck here. You're very generous, and I'm so appreciate.  If c-s receives un-normalized XYZ values, as described in #1, will it output reasonable xy values?  - J

**Reply by thomasmansencal** [reply]
Correct, because the CIE xy chromaticity coordinates are actually projected along the Luminance `Y` axis, their normalisation / scaling is not important! You can verify that easily:

```python
>>> import colour
>>> colour.XYZ_to_xy(colour.sd_to_XYZ(colour.SDS_ILLUMINANTS["D65"], n=683))
array([ 0.31271107,  0.32900848])
>>> XYZ = colour.sd_to_XYZ(colour.SDS_ILLUMINANTS["D65"], n=683)
>>> colour.XYZ_to_xy(XYZ / XYZ[1])
array([ 0.31271107,  0.32900848])
```
